doc/interfaces/registration_flow.rst

```diff
@@ -0,0 +1,172 @@
+``affine_reg_output``.
+
+In case you did not specify the output directory, the transformed image file
+and affine matrix would be saved in the input directory by default. If you did
+not specify the name of the output affine matrix, the affine matrix will be
+saved to a file named ``affine.txt`` by default, located in the output
+directory also by default.
+
+-----------------------
+Image Registration Flow
+-----------------------
+
+Image registration flow is a generic workflow which can be used to perform
+different types of registration such as center of mass, transformation, rigid
+body affine which includes translation, rotation, shearing and scaling. We can
+intend to use only one type of registration. Alternatively, using image
+registration, we can perform a registration in a progressive manner. For
+example, using affine registration with ``progressive`` set to ``True`` will
+involve center of mass, translation, rigid body and full affine registration.
+Whereas, when ``progressive`` is ``False`` the registration will include only
+center of mass and affine registration. The progressive registration will be
+slower but will improve the quality.
+
+We will first create a directory in which to save the transformed image and the
+affine matrix (e.g.: ``image_reg_output``)::
+
+    mkdir image_reg_output
+
+To run the image registration, we need to specify the paths to the static image
+file, and to the moving image file, followed by the optional arguments. In this
+case, we will be specifying the type of registration to be performed
+(``transform``) and the output directory (``out_dir``).
+
+To perform center of mass registration, we will call the ``dipy_align_affine``
+command with ``transform`` parameter set to ``com`` e.g.::
+
+    dipy_align_affine <path_to_static_file> <path_to_moving_file> --transform "com" --out_dir "image_reg_output"
+
+This command will save the transformed image and the affine matrix to the
+specified output directory.
+
+------------------------------------
+Symmetric Diffeomorphic Registration
+------------------------------------
+
+Symmetric Diffeomorphic Registration is performed using the ``Symmetric
+Normalization (SyN)`` algorithm proposed by Avants et al. [Avants09]_ (also
+implemented in the ANTs software [Avants11]_). It is an optimization technique
+that brings the moving image closer to the static image.
+
+Create a directory in which to save the transformed image (e.g.:
+``syn_reg_output``)::
+    
+    mkdir syn_reg_output
+
+To run the symmetric normalization registration method, we need to specify the
+paths to the static image file, and to the moving image file, followed by
+optional arguments. In this case, we will be specifying the metric (``metric``),
+the output directory (``out_dir``) and the file name of the output warped image
+(``out_warped``). You can use cc (Cross Correlation), ssd (Sum Squared
+Difference) or em (Expectation-Maximization) as metrics.
+
+We will run the command as::
```

Comment:
The symmetric diffeomorphic registration method in DIPY is run through the {double_backticks}dipy_align_syn{double_backticks} command, e.g.::

Reply:
Addressed in 2b43d64